### ISSUE TITLE
Error message from PAHM

### ISSUE THREAD

**Issue by yunfangsun** [noaa-ocs-modeling/PaHM#26]
Hi @pvelissariou1 @SorooshMani-NOAA,

Do you have any ideas on how to fix the following error in PAHM that I run into it:

---------- MODEL PARAMETERS ----------

InitLogging not called :: ProcessAsymmetricVortexData: 4 isotachs were nonzero.
InitLogging not called :: ProcessAsymmetricVortexData: 6 isotachs were nonzero.
InitLogging not called :: ProcessAsymmetricVortexData: 6 isotachs were nonzero.
InitLogging not called :: ProcessAsymmetricVortexData: 4 isotachs were nonzero.
InitLogging not called :: ^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@: The GAHM asymmetric data structure has more than 4 iSotachs in cycle 37.
_______________________________________________________________

Thank you!

Yunfang

**Comment by pvelissariou1** [comment]
@Yunfang Sun - NOAA Affiliate ***@***.***> What is the hurricane
you are working on? Can you share the input data (deck file) you are using?
I haven't experienced this error before.

Takis

Panagiotis Velissariou, Ph.D., P.E.
UCAR Scientist
National Ocean and Atmospheric Administration
National Ocean Service
Office of Coast Survey CSDL/CMMB
Physical Scientist - Project Lead
cell: (205) 227-9141
email: ***@***.***

**Comment by pvelissariou1** [comment]
Only four (4) isotachs are allowed per cycle.

Panagiotis Velissariou, Ph.D., P.E.
UCAR Scientist
National Ocean and Atmospheric Administration
National Ocean Service
Office of Coast Survey CSDL/CMMB
Physical Scientist - Project Lead
cell: (205) 227-9141
email: ***@***.***

**Comment by yunfangsun** [comment]
Hi Takis, @pvelissariou1 

I am working on the 2022 Hurricane Ian case, my input files are in the folder of /scratch2/STI/coastal/noscrub/Yunfang.Sun/run_schism/pahm on Hera.

Thank you!

Yunfang

**Comment by SorooshMani-NOAA** [comment]
@yunfangsun I'm not sure what that error is, but I have a couple of questions that might help find the answer: Are you using `stormevents` to download the track file? If so what advisory are you using?

One thing is that using `stormevents` for non-`BEST` tracks gives you a track file where the date for each advisory is the advisory issuance date, not the forecasted date. I'm not sure if that's a problem. In `BEST` track file you only get the hindcast dates.

I'm also tagging @FariborzDaneshvar-NOAA here as well as he will start using PaHM on SCHISM soon.

**Comment by yunfangsun** [comment]
@SorooshMani-NOAA 
Yes, I am using stormevents to download the track file.

The commands are as follows:

from stormevents.nhc import VortexTrack
track = VortexTrack.from_storm_name('ian', 2022)
storm_best_track = storm.track()
storm_best_track.to_file('hurricane-track.dat')

**Comment by SorooshMani-NOAA** [comment]
@pvelissariou1 please let me know if anything needs to change in `stormevents` when writing the `.dat` file for PaHM. Based on @yunfangsun's script, he's getting the best track, so the date issue described at https://github.com/noaa-ocs-modeling/PaHM/issues/26#issuecomment-1582481689 shouldn't be the issue here!

**Comment by pvelissariou1** [comment]
I am in the ADCIRC Meeting. I  will try to take a look at the data file,
most likely tomorrow. @yunfang have you tried to download the original and
use PAHM to create the output ATM data? PaHM does some basic filtering
before starting the calculations.

On Thursday, June 8, 2023, Soroosh Mani ***@***.***> wrote:
> @pvelissariou1 please let me know if anything needs to change in
stormevents when writing the .dat file for PaHM. Based on @yunfangsun's
script, he's getting the best track, so the date issue described at #26
(comment) shouldn't be the issue here!
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.<
https://ci5.googleusercontent.com/proxy/J-wBB8hA8WER6lxobIdpf_Gtwo6WPrDkyEiorkFgEHHXy3xu230XmpXolY6-lmk-ixegJYgcCCtwQ6fuuLJVzcBVoCXg3FbXqhnFPPXShQiMaIEhKDWnmazqe-hq1XnTO8VSqNDhFtAwtvhT15fi2x0kAniKT0fdjiaq81P9VDbtt0Tgfnj9RmNOgZmPnAAwmRVeLFnIUN6qEXrA4FuYWxQ1fLidfilPFv9MtdEmQoZlY4xMrw=s0-d-e1-ft#https://github.com/notifications/beacon/APC7TP2ODOCTBI4A6ZVUZXLXKHFY3A5CNFSM6AAAAAAY6SLJPSWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTS6KP4W6.gif>Message
ID: ***@***.***>

-- 
Panagiotis Velissariou, Ph.D., P.E.
UCAR Scientist
National Ocean and Atmospheric Administration
National Ocean Service
Office of Coast Survey CSDL/CMMB
Physical Scientist - Project Lead
cell: (205) 227-9141
email: ***@***.***

**Comment by yunfangsun** [comment]
Hi Takis, @pvelissariou1 

I will be on leave tomorrow, we can talk about it next week, and I will try the original PAHM to see how it works.

**Comment by yunfangsun** [comment]
On Mon, Jun 12, 2023 at 12:45 PM Panagiotis Velissariou - NOAA Affiliate <[panagiotis.velissariou@noaa.gov](mailto:panagiotis.velissariou@noaa.gov)> wrote:
Hi Yunfang,

I used your data file for Ian in PaHM. My mesh was set to ESTOFS-250m.
PaHM digested the file just fine.

Holland model:  no errors reported and it generated the output wind data just fine.

GAHM model:  errors reported for number of isotachs > 4
Offending lines:
AL, 09, 2022093018,   , BEST,   0, 333N,  792W,   70,  978, HU,  34, NEQ,  190,  130,  100,   60, 1007,  210,  40,  90,   0,   L,   0,    , 355,   9,        IAN,,,,,,,,
AL, 09, 2022093018,   , BEST,   0, 333N,  792W,   70,  978, HU,  50, NEQ,   80,   60,   80,   40, 1007,  210,  40,  90,   0,   L,   0,    , 355,   9,        IAN,,,,,,,,
AL, 09, 2022093018,   , BEST,   0, 333N,  792W,   70,  978, HU,  64, NEQ,    0,   30,   40,    0, 1007,  210,  40,  90,   0,   L,   0,    , 355,   9,        IAN,,,,,,,,
AL, 09, 2022093018, 05, BEST,   0, 333N,  792W,   70,  978, HU,  34, NEQ,  190,  130,  100,   60, 1007,  210,  40,   0,   0,   L,   0,    ,   0,   0,        IAN,,,,,,,,
AL, 09, 2022093018, 05, BEST,   0, 333N,  792W,   70,  978, HU,  50, NEQ,   80,   60,   80,   40, 1007,  210,  40,   0,   0,   L,   0,    ,   0,   0,        IAN,,,,,,,,
AL, 09, 2022093018, 05, BEST,   0, 333N,  792W,   70,  978, HU,  64, NEQ,    0,   30,   40,    0, 1007,  210,  40,   0,   0,   L,   0,    ,   0,   0,        IAN,,,,,,,,
So, basically the 3 "red" lines are duplicates of the 3 "green" lines. Delete the 3 "red" lines in the file and you should be ok.

I am not sure that if this is a rare occurrence or, we need to filter out the extra entries based on some criteria. I'll investigate further.
It is the first time I encountered this type of error.
In SCHISM, the PaHM code is identical to the standalone PaHM so, it shouldn't be an issue from SCHISM's side.

SCHISM, by default uses the GAHM model (hardcoded in schism/src/Core/PaHM/PaHM_Utilities.F90, modelType=10 - line 266).
You might want to change this by setting modelType=1 to use the Holland model.

Takis

Panagiotis Velissariou, Ph.D., P.E.
UCAR Scientist
National Ocean and Atmospheric Administration
National Ocean Service
Office of Coast Survey CSDL/CMMB
Physical Scientist - Project Lead
cell: (205) 227-9141
email: [panagiotis.velissariou@noaa.gov](mailto:panagiotis.velissariou@noaa.gov)
